File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/access/annotationprovider/BotAddedAnnotationProvider.java
package ru.tggc.telegrambotcore.access.annotationprovider;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.handle.BotAddedHandle;
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta;

import java.lang.reflect.Method;

@Component
public class BotAddedAnnotationProvider implements AnnotationProvider {

    @Override
    public boolean supports(Method m) {
        return m.isAnnotationPresent(BotAddedHandle.class);
    }

    @Override
    public HandleMeta extractMeta(Method m) {
        return HandleMeta.fromDefault();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/access/annotationprovider/BotAddedAnnotationProvider.kt
package ru.tggc.telegrambotcore.access.annotationprovider

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.handle.BotAddedHandle
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta
import java.lang.reflect.Method

@Component
class BotAddedAnnotationProvider : AnnotationProvider {
    override fun supports(m: Method): Boolean = m.isAnnotationPresent(BotAddedHandle::class.java)

    override fun extractMeta(m: Method): HandleMeta? = HandleMeta.fromDefault()
}
